src/psyclone/psyir/nodes/node.py

```diff
@@ -894,25 +894,41 @@ def position(self):
     @property
     def abs_position(self):
         '''
-        Find a Node's absolute position in the tree (starting with 0 if
-        it is the root). Needs to be computed dynamically from the
-        starting position (0) as its position may change.
+        Find a Node's absolute position in the tree (starting with
+        START_POSITION if it is the root). Needs to be computed dynamically
+        from the root as its position may change.
 
-        :returns: absolute position of a Node in the tree
+        :returns: absolute position of a Node in the tree.
         :rtype: int
 
-        :raises InternalError: if the absolute position cannot be found
         '''
-        from psyclone.psyir.nodes import Schedule
-        if self.root == self and isinstance(self.root, Schedule):
+        if self.root == self:
             return self.START_POSITION
-        found, position = self._find_position(self.root.children,
-                                              self.START_POSITION)
-        if not found:
-            raise InternalError("Error in search for Node position "
-                                "in the tree")
+        _, position = self._find_position(self.root.children,
+                                          self.START_POSITION)
         return position
 
+    @property
+    def abs_position_in_routine(self):
```

Comment:
I realized this new method was not needed because the functionality is already provided by the _find_position() method. So I removed the abs_position_in_routine method now.